decorator.js

```diff
@@ -21,11 +21,11 @@ function addClickOutsideListener(Component) {
     },
 
     render: function render() {
-      return React.createElement(Component, objectAssign({
+      return React.createElement(Component, objectAssign({}, this.props, {
```

Comment:
The enableOnClickOutside and disableOnClickOutside methods need to be added by the HOC to the properties passed to the component, or the component won't have access to them. If you replaced lines 24-28 with your snippet, then you wouldn't be able to enable/disable listening from the wrapped component.

Unless I'm misunderstanding your question?